src/pagination/pagination.js

```diff
@@ -0,0 +1,194 @@
+  };
+
+  state = {isMenuOpen: false};
+
+  dropdownContainerRef = React.createRef();
+
+  onPageClick = (event: MouseEvent) => {
+    const el: ?HTMLDivElement = this.dropdownContainerRef.current;
+    /* eslint-disable-next-line flowtype/no-weak-types */
+    if (el && !el.contains((event.target: any))) {
+      this.setState({isMenuOpen: false});
+    }
+  };
+
+  getMenuOptions = memoize(numPages => {
+    const menuOptions = [];
+    for (let i = 1; i <= numPages; i++) {
+      menuOptions.push(({label: i}: MenuItemT));
+    }
+    return menuOptions;
+  });
+
+  onMenuItemSelect = (item: MenuItemT) => {
+    const {onPageChange, currentPage} = this.props;
+    const page = item.label;
+    if (page !== currentPage) {
+      onPageChange && onPageChange(page, currentPage);
+      this.onDropdownButtonClick();
+    }
+  };
+
+  onPrevClick = (...args: *) => {
+    const {currentPage, onPageChange, onPrevClick} = this.props;
+    onPageChange && onPageChange(currentPage - 1, currentPage);
+    onPrevClick && onPrevClick(args);
+  };
+
+  onNextClick = (...args: *) => {
+    const {currentPage, onPageChange, onNextClick} = this.props;
+    onPageChange && onPageChange(currentPage + 1, currentPage);
+    onNextClick && onNextClick(args);
+  };
+
+  onDropdownButtonClick = () => {
+    const isMenuOpen = !this.state.isMenuOpen;
+    // no __BROWSER__ check because click only happens client-side
+    if (isMenuOpen) {
+      document.addEventListener('click', this.onPageClick, {
+        capture: true,
+      });
+    } else {
+      document.removeEventListener('click', this.onPageClick, {
+        capture: true,
+      });
+    }
+    this.setState({isMenuOpen});
+  };
+
+  render() {
+    const {
+      overrides = {},
+      currentPage,
+      prepositionLabel = '',
+      numPages,
+    } = this.props;
+    const {isMenuOpen} = this.state;
+
+    const {component: Root, props: rootProps} = getOverrideObject(
+      overrides.Root,
+      StyledRoot,
+    );
+    const {component: PrevButton, props: prevButtonProps} = getOverrideObject(
+      overrides.PrevButton,
+      StyledBaseButton,
+    );
+    const {component: NextButton, props: nextButtonProps} = getOverrideObject(
+      overrides.NextButton,
+      StyledBaseButton,
+    );
+    const {component: MaxLabel, props: maxLabelProps} = getOverrideObject(
+      overrides.MaxLabel,
+      StyledMaxLabel,
+    );
+    const {
+      component: DropdownContainer,
+      props: dropdownContainerProps,
+    } = getOverrideObject(overrides.DropdownContainer, StyledDropdownContainer);
+    const {
+      component: DropdownButton,
+      props: dropdownButtonProps,
+    } = getOverrideObject(overrides.DropdownButton, StyledDropdownButton);
+    const {
+      component: DropdownMenu,
+      props: dropdownMenuProps,
+    } = getOverrideObject(overrides.DropdownMenu, Menu);
+
+    const options = this.getMenuOptions(numPages);
+
+    return (
+      <Root {...rootProps}>
+        <Button
```

Comment:
To the first question for a discussion. What if we abstract it into a `PrevButton` subcomponent instead? It will 
 make the overrides more complex, but will provide the way to override/replace the whole `Button` component used.

Reply:
This is worth exploring for sure, maybe convention is to allow top-level override of all internal components and if users want to do sub `overrides` they can do that manually.